foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/http/HttpUtils.java

```diff
@@ -94,6 +97,14 @@ public static String encodePathParam(String pathParam) {
     return UrlEscapers.urlPathSegmentEscaper().escape(pathParam);
   }
 
+  public static String decodePathParam(String pathParam) throws UnsupportedEncodingException {
+    String res = uriDecodePath(pathParam);
+    if (StringUtils.isEmpty(res)) {
+      return URLDecoder.decode(pathParam, "UTF-8");
+    }
+    return res;
+  }
+
   public static String uriDecodePath(String path) {
     if (path == null) {
```

Comment:
This function seems not correct. According to the java docs, URL(path) will not parse the path, users should use 
```new URI(null, null, path, null);``` to parse the parse

Reply:
use `new URI(null, null, path, null);` will cause ut failed:
org.apache.servicecomb.common.rest.codec.param.TestPathProcessor#testGetSpaceEncoded

`new URI(null, null, path, null);` can not convert '%20' to ' '